grazer/src/main/java/edu/asu/diging/grazer/web/PersonController.java

```diff
@@ -64,13 +65,27 @@ public int compare(WikidataStatement o1, WikidataStatement o2) {
         return "person";
     }
     
-    @RequestMapping("/concept/{personId}/graph")
-    public String getPersonGraph(@PathVariable("personId") String personId, Model model) {
+    public boolean sameEdge(Edge edge1, Edge edge2){
+        return edge1.getSourceNode().getLabel().equals(edge2.getSourceNode().getLabel()) && 
+                edge1.getTargetNode().getLabel().equals(edge2.getTargetNode().getLabel());
+     }
+    
+    @RequestMapping("/concept/{conceptId}/graph")
+    public String getPersonGraph(@PathVariable("conceptId") String conceptId, Model model) {
         
-        IConcept concept = cache.getConceptById(personId);
+        IConcept concept = cache.getConceptById(conceptId);
+        List<Edge> edgeList = graphDbConnector.getEdges(concept.getUri());
+        List<Edge> duplicates = new ArrayList<Edge>();
+        for(int i = 0; i < edgeList.size()-1; i++) {
+            for(int j = i+1; j < edgeList.size(); j++) {
+                if(sameEdge(edgeList.get(i), edgeList.get(j))) {
+                    duplicates.add(edgeList.get(j));
+                }
+            }
+        }
```

Comment:
instead of a loop in a loop (which is usually not the best idea performance-wise), while iterating just remember what node combinations you've already looked at and then check against that list (e.g. have a list of strings uri1_uri2 then to a contains with the current edge).